### PR TITLE
feat(oidc): Add OIDC provider condition

### PR DESCRIPTION
**Pull request by Balsir**
# Description

Adds condition for creation of OIDC provider.

## Type of change

- [ ] A bug fix (PR prefix `fix`)
- [x] A new feature (PR prefix `feat`)
- [ ] A code change that neither fixes a bug nor adds a feature (PR prefix `refactor`)
- [ ] Adding missing tests or correcting existing tests (PR prefix `test`)
- [ ] Changes that do not affect the meaning of the code like white-spaces, formatting, missing semi-colons, etc. (PR prefix `style`)
- [ ] Changes to our CI configuration files and scripts (PR prefix `ci`)
- [ ] Documentation only changes (PR prefix `docs`)

### CODE REVIEW

File: modules/addon-oidc/iam.tf [hunk 1]
```diff
@@ -1,14 +1,14 @@
 locals {
+  oidc_provider_create     = var.enabled == true && var.oidc_provider_create == true #  var.oidc_provider_create == true
```
Comment (jaygridley) [inline modules/addon-oidc/iam.tf:2]: Left over comment?

File: modules/addon-oidc/iam.tf [hunk 2]
```diff
@@ -47,7 +47,7 @@ data "aws_iam_policy_document" "this_oidc" {
 
     principals {
       type        = "Federated"
-      identifiers = [try(aws_iam_openid_connect_provider.this[0].arn, var.oidc_custom_provider_arn)]
+      identifiers = [coalesce(var.oidc_custom_provider_arn, try(aws_iam_openid_connect_provider.this[0].arn, ""))]
```
Comment (jaygridley) [inline modules/addon-oidc/iam.tf:50]:
```suggestion
      identifiers = [coalesce(var.oidc_custom_provider_arn, one(aws_iam_openid_connect_provider.this[*].arn))]
```

File: modules/addon-oidc/outputs.tf [hunk 3]
```diff
@@ -1,5 +1,10 @@
+output "oidc_provider_enabled" {
+  description = "Whether oidc provider is enabled"
```
Comment (jaygridley) [inline modules/addon-oidc/outputs.tf:2]:
```suggestion
  description = "Whether OIDC provider is enabled"
```

File: modules/addon-oidc/outputs.tf [hunk 4]
```diff
@@ -1,5 +1,10 @@
...
+  value       = local.oidc_provider_create
+}
+
 output "oidc_role_enabled" {
-  description = "Whether is oidc role enabled"
+  description = "Whether oidc role is enabled"
```
Comment (jaygridley) [inline modules/addon-oidc/outputs.tf:7]:
```suggestion
  description = "Whether OIDC role is enabled"
```

File: modules/addon-oidc/variables.tf [hunk 5]
```diff
@@ -6,6 +6,12 @@ variable "enabled" {
 
 # ================ oidc variables (optional) ================
 
+variable "oidc_provider_create" {
+  type        = bool
+  default     = null
+  description = "Whether to create oidc provider. Defaults to `true`. Set to false if you want to disable default oidc provider when oidc_custom_provider_arn is set"
```
Comment (jaygridley) [inline modules/addon-oidc/variables.tf:12]: Se to the same description as in `variables-addon-oidc.tf`.

File: modules/addon-oidc/variables.tf [hunk 6]
```diff
@@ -105,5 +111,5 @@ variable "oidc_openid_thumbprints" {
 variable "oidc_custom_provider_arn" {
   type        = string
   default     = null
-  description = "Specifies a custom OIDC provider ARN. If provided, the module will not create a default OIDC provider. Defaults to `\"\"`."
+  description = "Specifies a custom OIDC provider ARN. If specified, overrides provider created by this module. If set, it is recommended to disable default OIDC provider creation by setting var.oidc_provider_create to false. Defaults to `\"\"`."
```
Comment (jaygridley) [inline modules/addon-oidc/variables.tf:114]: Se to the same description as in `variables-addon-oidc.tf`.

File: modules/addon-oidc/iam.tf (hunk 1 above)
Comment (Balsir) [reply modules/addon-oidc/iam.tf]: gone

File: modules/addon-oidc/variables.tf (hunk 6 above)
Comment (Balsir) [reply modules/addon-oidc/variables.tf:114]: Should now be fixed

File: modules/addon-oidc/variables.tf (hunk 5 above)
Comment (Balsir) [reply modules/addon-oidc/variables.tf]: Should now be fixed

File: variables-addon-oidc.tf [hunk 7]
```diff
@@ -97,14 +86,26 @@ variable "oidc_assume_role_policy_condition_variable" {
...
   default     = null
-  description = "Specifies a custom OIDC provider ARN. If specified, overrides provider created by this module. If set, it is recommended to disable default OIDC provider creation by setting var.oidc_provider_create to false. Defaults to `\"\"`."
+  description = "OIDC provider URL. Defaults to `\"\"`."
 }
 
-variable "oidc_tags" {
-  type        = map(string)
+variable "oidc_openid_thumbprints" {
+  type        = list(string)
+  default     = []
```
Comment (jaygridley) [inline variables-addon-oidc.tf:103]:
```suggestion
  default     = null
```

File: variables-addon-oidc.tf [hunk 8]
```diff
@@ -97,14 +86,26 @@ variable "oidc_assume_role_policy_condition_variable" {
   description = "Specifies the variable to use for the assume role trust policy. Defaults to `\"\"`."
 }
 
-variable "oidc_custom_provider_arn" {
+variable "oidc_openid_client_ids" {
+  type        = list(string)
+  default     = []
```
Comment (jaygridley) [inline variables-addon-oidc.tf:91]:
```suggestion
  default     = null
```

File: variables-addon-oidc.tf [hunk 9]
```diff
@@ -34,60 +35,48 @@ variable "oidc_policy_enabled" {
...
-  default     = null
-  description = "Specifies the condition test to use for the assume role trust policy. Defaults to `StringLike`."
+  default     = "StringLike"
+  description = "Specifies the condition test to use for the assume role trust policy. Defaults to `StringEquals`."
 }
 
 variable "oidc_assume_role_policy_condition_values" {
   type        = list(string)
-  default     = null
+  default     = []
```
Comment (jaygridley) [inline variables-addon-oidc.tf:79]:
```suggestion
  default     = null
```

File: variables-addon-oidc.tf [hunk 10]
```diff
@@ -34,60 +35,48 @@ variable "oidc_policy_enabled" {
...
   default     = null
-  description = "List of thumbprints of the OIDC provider's server certificate. Defaults to `[]`."
+  description = "OIDC resources tags. Defaults to `{}`."
 }
 
 variable "oidc_assume_role_policy_condition_test" {
   type        = string
-  default     = null
-  description = "Specifies the condition test to use for the assume role trust policy. Defaults to `StringLike`."
+  default     = "StringLike"
```
Comment (jaygridley) [inline variables-addon-oidc.tf:73]:
```suggestion
  default     = null
```

File: variables-addon-oidc.tf [hunk 11]
```diff
@@ -1,28 +1,29 @@
 # IMPORTANT: This file is synced with the "terraform-aws-eks-universal-addon" module. Any changes to this file might be overwritten upon the next release of that module.
 
 # ================ oidc variables (optional) ================
+
 variable "oidc_provider_create" {
   type        = bool
   default     = null
-  description = "Whether to create oidc provider. Defaults to `true`. Set to false if you want to disable default oidc provider when oidc_custom_provider_arn is set"
+  description = "Whether to create OIDC provider. Defaults to `true`. Set to false if you want to disable default OIDC provider when oidc_custom_provider_arn is set"
```
Comment (jaygridley) [inline variables-addon-oidc.tf:8]:
```suggestion
  description = "Whether to create OIDC provider. Defaults to `true`. Set to false if you want to disable default OIDC provider when oidc_custom_provider_arn is set."
```

**Review by jaygridley — CHANGES_REQUESTED** [review]

**Review by jaygridley — APPROVED** [review]
LGTM

**Pull request merged**